### PR TITLE
New post hashtag and account issues

### PR DESCRIPTION
**Pull request by JohnOberhauser**
There were bugs with highlighting, searching, and replacing hashtags and accounts in the new post screen.  The bugs stemmed from the algorithm not taking into consideration new line characters.

- fixed issues with highlighting, searching, and replacing hashtags and accounts
- added some extra unit tests

### CODE REVIEW

File: core/common/src/main/java/org/mozilla/social/common/utils/TextFieldValueExtensions.kt [hunk 1]
```diff
@@ -47,29 +52,39 @@ fun TextFieldValue.replaceSymbolText(
     newText: String,
 ): TextFieldValue {
     if (!text.contains(symbol)) return this
-    val firstSpaceIndexAfterCursor = text.indexOf(' ', startIndex = selection.end)
+    val firstSpaceOrNewLineIndexAfterCursor = getFirstSpaceOrNewLineIndexAfterCursor(
```
Comment (devotaaabel) [inline core/common/src/main/java/org/mozilla/social/common/utils/TextFieldValueExtensions.kt:55]: Can you add kdoc to this function, and others in this file that might be a little nebulous? It's hard to tell what this does. It seems to be replacing the text after the symbol, but with what, it's hard to parse. Is it updating the text style? Maybe there's a clearer way to name this function?

Comment (JohnOberhauser) [reply core/common/src/main/java/org/mozilla/social/common/utils/TextFieldValueExtensions.kt:67]: updated.  Let me know if it's clear enough

Comment (devotaaabel) [reply core/common/src/main/java/org/mozilla/social/common/utils/TextFieldValueExtensions.kt:67]: Much better

**Review by devotaaabel — APPROVED** [review]
